### PR TITLE
chore: [Foss] Sync public: Add retry and logging for dlq/http/cors flaky requests

### PR DESCRIPTION
**Pull request by mingkun2020**
*Description of changes:*
add retry and logging for dlq/http/cors flaky requests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: integration/combination/test_api_with_gateway_responses.py [hunk 1]
```diff
@@ -1,9 +1,12 @@
+import logging
 from unittest.case import skipIf
 
+from tenacity import stop_after_attempt, retry_if_exception_type, after_log, wait_exponential, retry, wait_random
```
Comment (jfuss) [inline integration/combination/test_api_with_gateway_responses.py:4]: Where is tenacity coming from?

**Review by jfuss — CHANGES_REQUESTED** [review]

File: integration/combination/test_api_with_gateway_responses.py (hunk 1 above)
Comment (mingkun2020) [reply integration/combination/test_api_with_gateway_responses.py:4]: [Tenacity](https://pypi.org/project/tenacity/) is a retry library we used in integration tests and it has Apache 2.0 liciense.

File: integration/combination/test_api_with_gateway_responses.py [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+
 from integration.helpers.base_test import BaseTest
-from integration.helpers.deployer.utils.retry import retry
 from integration.helpers.resource import current_region_does_not_support
 from integration.config.service_names import GATEWAY_RESPONSES
 
+LOG = logging.getLogger(__name__)
```
Comment (mndeveci) [inline integration/combination/test_api_with_gateway_responses.py:10]: It seems that we added this logger but I can't see any logging output, is this redundant?

Comment (mingkun2020) [reply integration/combination/test_api_with_gateway_responses.py:10]: It is used in line 44.

Comment (mndeveci) [reply integration/combination/test_api_with_gateway_responses.py:10]: Oh I see, library uses it

**Review by mndeveci — APPROVED** [review]

**Review by jfuss — APPROVED** [review]

**Pull request merged**